src/app/InteractionModelEngine.h

```diff
@@ -402,7 +403,10 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
     }
 #endif
 
-    DataModel::Provider * GetDataModelProvider() const;
+    // Temporarly NOT const because the data model provider will be auto-set
+    // to codegen on first usage. This behaviour will be changed once each
+    // application must explicitly set the data model provider.
+    DataModel::Provider * GetDataModelProvider();
```

Comment:
No, however this is current development and I figured we would iterate on it anyway (I have todos to make data model setting be a choice of the app, so app builds its backend (e.g. ember is currently on every app and every app has its own zapfile) and then provided should be set in main. Then this can become const.

Regarding issues, we currently have 1635 open issues (and 10K closed, a lot by stale bot) so for my own development adding github issues does not seem helpful. I can add an issue if you think it is important, it is just my default to not add them if actively developing things.

Reply:
SGTM